ctms/crud.py

```diff
@@ -267,6 +267,63 @@ def get_contacts_by_any_id(
     return data
 
 
+def get_all_acoustic_records_before(
+    db: Session,
+    end_time: datetime,
+    retry_limit: int = 5,
+) -> List[PendingAcousticRecord]:
+    """
+    Get all the pending records before a given date. Allows retry limit to be provided at query time."""
+    pending_records: List[PendingAcousticRecord] = (
+        db.query(PendingAcousticRecord)
+        .filter(
+            PendingAcousticRecord.update_timestamp < end_time,
+            PendingAcousticRecord.retry < retry_limit,
+        )
+        .order_by(asc(PendingAcousticRecord.update_timestamp))
+        .all()
+    )
+
+    return pending_records
+
+
+def get_acoustic_record_as_contact(
+    record: PendingAcousticRecord,
+) -> ContactSchema:
+    # if list to list conversion desired this function
+    # can be used with map(get_acoustic_record_as_contact, record_list)
+    contact: ContactSchema = ContactSchema.parse_obj(
+        {
+            "amo": record.email.amo,
+            "email": record.email,
+            "fxa": record.email.fxa,
+            "mofo": record.email.mofo,
+            "newsletters": record.email.newsletters,
+            "vpn_waitlist": record.email.vpn_waitlist,
+        }
+    )
+    return contact
+
+
+def schedule_acoustic_record(
+    db: Session,
+    email_id: UUID4,
+) -> None:
+    db_pending_record = PendingAcousticRecord(email_id=email_id)
+    db.add(db_pending_record)
+
+
+def retry_acoustic_record(db: Session, pending_record: PendingAcousticRecord) -> None:
+    pending_record.retry += 1
+    pending_record.update_timestamp = datetime.now(timezone.utc)
+    # db.commit() # TODO: will this be done by the background job?
+
+
+def delete_acoustic_record(db: Session, pending_record: PendingAcousticRecord) -> None:
+    db.delete(pending_record)
+    # db.commit() # TODO: will this be done by the background job?
```

Comment:
I'm not too sure if I need to call a commit here, or if the sessions should be committed by the background job.

Maybe just `retry` and `delete` would need this commit? And since it's only called in the background job, it could be a commit after all deletes, or after each delete?

Not sure what the preference would be here...